Automattic/woocommerce-payments
# Replace direct database queries in WCPay Subscription code which pulls orders and subscriptions via invoice or subscription id

Closes #4147 

#### Changes proposed in this Pull Request

This PR is part of our efforts to update Subscriptions and related functionality to support HPOS environments. 

In WCPayments there are 2 places where we were querying the database directly to pull orders and subscriptions with a specific meta key and value.

This PR replaces those with `wc_get_orders()`. 

In `subscriptions-core` we introduced a helper function to do something similar, however we can't use it here because we can't guarantee that sites with WC Subscriptions have that function. 

#### Testing instructions

There are pretty extensive unit test coverage of this area. 
To test I:

1. Purchased a WCPay subscription
     - ⚠️  requires https://github.com/Automattic/woocommerce-subscriptions-core/pull/258
2. With WP-Console (plugin) I mimicked calling `get_subscription_from_wcpay_subscription_id()`. See code snippet example below.

I repeated these steps on the following environments and confirmed that the subscription ID was returned each time:
- HPOS (no sync) 👍 
- HPOS (w sync) 👍 
- WP Posts (no sync) 👍 
- WP posts (w sync) 👍 

```php
// Replace this subscription ID with the new subscription purchased on different environments.
$subscription = WC_Payments_Subscription_Service::get_subscription_from_wcpay_subscription_id( 'sub_1M3xEhCSQIDdUhVAkTMveWbf' );

if ( $subscription ) {
    echo $subscription->get_id();
} else {
    echo 'No subscription found';
}
```




-------------------


**Post merge**
